tests/test_contracts.py

```
@@ -170,22 +185,249 @@ def test_decode_logs(self):
         # Test ignoring logs from other contracts
-        log_mock_other = MagicMock()
-        log_mock_other.address = "0xOther"
-        receipt.logs = [log_mock_other]
+        log_entry_other = {"address": "0xOther"}
+        receipt = {"logs": [log_entry_other]}
         logs = contract._decode_logs(receipt)
         assert len(logs) == 0
 
+    @pytest.mark.usefixtures("mock_w3", "mock_load_abi")
+    def test_init_load_dotenv_restores_rpc_url(self):
+        """When RPC_URL is unset, load_dotenv can populate it (lines 76–79)."""
```

Ruff (0.15.10)

[warning] Docstring contains ambiguous `–` (EN DASH). Did you mean `-` (HYPHEN-MINUS)?

(RUF002)